pyteal/ast/router.py

```diff
@@ -0,0 +1,568 @@
+from dataclasses import dataclass, field, fields, astuple
+from typing import cast, Optional
+from enum import IntFlag
+
+import algosdk.abi as sdk_abi
+
+from pyteal.config import METHOD_ARG_NUM_LIMIT
+from pyteal.errors import TealInputError, TealInternalError
+from pyteal.types import TealType
+from pyteal.compiler.compiler import compileTeal, DEFAULT_TEAL_VERSION, OptimizeOptions
+from pyteal.ir.ops import Mode
+
+from pyteal.ast import abi
+from pyteal.ast.subroutine import (
+    OutputKwArgInfo,
+    SubroutineFnWrapper,
+    ABIReturnSubroutine,
+)
+from pyteal.ast.cond import Cond
+from pyteal.ast.expr import Expr
+from pyteal.ast.app import OnComplete, EnumInt
+from pyteal.ast.int import Int
+from pyteal.ast.seq import Seq
+from pyteal.ast.methodsig import MethodSignature
+from pyteal.ast.naryexpr import And, Or
+from pyteal.ast.txn import Txn
+from pyteal.ast.return_ import Approve
+
+
+################################################################################
+#                                  DISCLAIMER                                  #
```

Comment:
@ahangsu and reviewers - Where is the most appropriate place (or places) to add a disclaimer such that the library user is _unlikely_ to miss it?

My as is concern:  A library user won't see the disclaimer when reviewing data type docstrings.  In order to see the disclaimer, the library user must inspect source code.

I understand a general disclaimer was added because `pyteal/ast/router.py` is generally subject to same guarantees.

I'm open to suggestions on how to more directly present the disclaimer.  My 2c:
* Since `Router` is the usage entry point, perhaps adding to `Router` docstring is sufficient?
* Alternatively, if it still feels ambiguous, then I'd favor adding an explicit docstring note to each class covered by the disclaimer.

Reply:
+1 to adding the warning to `Router`'s docstring, since all other classes revolve around it.

Reply:
ok, lets move this into the top of `Router`'s docstring, and see how it fits there.

Reply:
@ahangsu As of https://github.com/algorand/pyteal/pull/170/commits/d37e00f0f99f8bbc3f7544a9c30eff4d6f325dab, the change to `Router` docstring looks good to me - thanks for making the change.

A few notes:
* From my perspective, I think the thread is resolved.  If others feel there's more to do, let's reopen.
* To @tzaffi's point, I opened a story requesting a user guide and explicitly requested defining level of support:  https://github.com/algorand/pyteal/issues/367 - _ABI: Prep user guide for initial ABI release_.